### PR TITLE
Add a gizmo menu to nuke

### PR DESCRIPTION
**Pull request by ClementHector**
## Brief description
Adding a menu gizmo in nuke with the same logic as scriptmenu.

## Description
The menu gizmo must make it possible to quickly implement gizmos in one or more menus without having to produce code in menu.py or init.py.

## Testing notes:
1. Open OpenPype Project settings
2. Go to project_settings/nuke/gizmo/
3. Fill the settings
image
4. You can click on the newly added gizmo
image

### CODE REVIEW

**Review by m-u-r-p-h-y — CHANGES_REQUESTED** [review]
Nuke startup will end with some errors

```
Nuke 12.2v9, 64 bit, built Sep  3 2021.
Copyright (c) 2021 The Foundry Visionmongers Ltd.  All Rights Reserved.
Licence expires on: 2022/7/11
D:\REPO\OpenPype\openpype\vendor\python\python_2\pkg_resources\py2_warn.py:19: UserWarning: ************************************************************
You are running Setuptools on Python 2, which is no longer
supported and
>>> SETUPTOOLS WILL STOP WORKING <<<
in a subsequent release. Please ensure you are installing
Setuptools using pip 9.x or later or pin to `setuptools<45`
in your environment.
If you have done those things and are still encountering
this message, please comment in
https://github.com/pypa/setuptools/issues/1458
about the steps that led to this unsupported combination.
************************************************************
  sys.version_info < (3,) and warnings.warn("*" * 60 + msg + "*" * 60)
  File "D:\REPO\OpenPype\openpype\hosts\nuke\startup/menu.py", line 65
    log.warning(f"This environment variable doesn't exist: {e}")
                                                              ^
SyntaxError: invalid syntax
```

image

**Comment by ClementHector** [comment]
Sorry for forgetting!

**Comment by jakubjezek001** [comment]
hey @ClementHector , I wonder if we could change the way Settings are designed? I have done little mockup here I hope it will make sense to you. 

image

Changes to the code will be added to proposed changes.

File: openpype/settings/entities/schemas/projects_schema/schemas/schema_nuke_scriptsgizmo.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (jakubjezek001) [inline openpype/settings/entities/schemas/projects_schema/schemas/schema_nuke_scriptsgizmo.json:1]:
```suggestion
{
    "type": "list",
    "key": "gizmo",
    "label": "Gizmo Menu",
    "is_group": true,
    "use_label_wrap": true,
    "object_type": {
        "type": "dict",
        "children": [
            {
                "type": "text",
                "key": "toolbar_menu_name",
                "label": "Toolbar Menu Name"
            },
            {
                "type": "path",
                "key": "gizmo_path",
                "label": "Gizmo directory path",
                "multipath": true,
                "multiplatform": true
            },
            {
                "type": "collapsible-wrap",
                "label": "Options",
                "collapsible": true,
                "collapsed": true,
                "children": [
                    {
                        "type": "path",
                        "key": "toolbar_icon_path",
                        "label": "Toolbar Icon Path",
                        "multipath": false,
                        "multiplatform": true
                    },
                    {
                        "type": "splitter"
                    },
                    {
                        "type": "list",
                        "key": "gizmo_definition",
                        "label": "Gizmo definitions",
                        "use_label_wrap": true,
                        "object_type": {
                            "type": "dict-conditional",
                            "enum_key": "sourcetype",
                            "enum_label": "Type of usage",
                            "enum_children": [
                                {
                                    "key": "python",
                                    "label": "Python",
                                    "children": [
                                        {
                                            "type": "text",
                                            "key": "title",
                                            "label": "Title"
                                        },
                                        {
                                            "type": "text",
                                            "key": "command",
                                            "label": "Python command"
                                        }
                                    ]
                                },
                                {
                                    "key": "file",
                                    "label": "File",
                                    "children": [
                                        {
                                            "type": "text",
                                            "key": "title",
                                            "label": "Title"
                                        },
                                        {
                                            "type": "text",
                                            "key": "file_name",
                                            "label": "Gizmo file name"
                                        }
                                    ]
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```

Comment (ClementHector) [reply openpype/settings/entities/schemas/projects_schema/schemas/schema_nuke_scriptsgizmo.json:1]: I made some tweaks and refactored based on this thinking.

**Review by mkolar — APPROVED** [review]
Nice work. The new settings are looking good

**Pull request merged**